runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -445,6 +445,151 @@ TR::Register * caseConversionHelper(TR::Node* node, TR::CodeGenerator* cg, bool
    return node->getRegister();
    }
 
+extern TR::Register *
+inlineFindElementFromArray(TR::Node * node, TR::CodeGenerator * cg)
+   {
+   cg->generateDebugCounter("z13/simd/indexOf", 1, TR::DebugCounter::Free);
+
+   TR::Register* address = cg->evaluate(node->getChild(1));
+   TR::Register* value = cg->evaluate(node->getChild(2));
+   TR::Register* index = cg->evaluate(node->getChild(3));
+   TR::Register* size = cg->gprClobberEvaluate(node->getChild(4));
+
+   // load length isn't used after loop, size must is adjusted to become bytes left
+   TR::Register* loopCounter = size;
+   TR::Register* loadLength = cg->allocateRegister();
+   TR::Register* indexRegister = cg->allocateRegister();
+   TR::Register* offsetAddress = cg->allocateRegister();
+   TR::Register* scratch = offsetAddress;
+
+   TR::Register* charBufferVector = cg->allocateRegister(TR_VRF);
+   TR::Register* resultVector = cg->allocateRegister(TR_VRF);
+   TR::Register* valueVector = cg->allocateRegister(TR_VRF);
+
+   TR::LabelSymbol* loopLabel = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* fullVectorLabel = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* notFoundInResidue = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* foundLabel = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* foundLabelExtractedScratch = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* failureLabel = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* doneLabel = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+
+   bool isCompressed = node->getSymbol()->castToMethodSymbol()->getMethod()->parmType(1) == TR::Int8;
+   const int elementSizeMask = isCompressed ? 0x0 : 0x1;   // byte or halfword mask
+   uintptrj_t headerSize = TR::Compiler->om.contiguousArrayHeaderSizeInBytes();
+   const int8_t sizeOfVector = 16;
```

Comment:
There is a `getVRFSize` API in the Machine class for this. Let's not hardcode it.

Reply:
b6c5709